lib/model/model.go

```diff
@@ -2882,16 +2882,30 @@ func (m *model) Availability(folder string, file protocol.FileInfo, block protoc
 	}
 	defer snap.Release()
 
-	return m.availabilityInSnapshotRLocked(cfg, snap, file, block), nil
+	if devices := m.availabilityInSnapshot(cfg, snap, file); len(devices) > 0 {
+		return devices, nil
+	}
+
+	return m.availabilityFromTemporaryRLocked(cfg, file, block), nil
```

Comment:
You don't actually hold the read lock here. You likely want to acquire it before calling both `availability` functions, and call the `...RLocked` variant of those in both cases.